src/ragas/experimental/testset/simulators/base.py

```diff
@@ -54,3 +56,23 @@ async def generate_scenarios(
     @abstractmethod
     async def generate_sample(self, scenario: Scenario) -> BaseEvalSample:
         pass
+
+    def get_prompts(self) -> t.Dict[str, Prompt]:
+        prompts = {}
+        for name, value in inspect.getmembers(self):
+            if isinstance(value, Prompt):
+                prompts.update({name: value})
+        return prompts
+
+    def set_prompts(self, **prompts):
+        available_prompts = self.get_prompts()
+        for key, value in prompts.items():
+            if key not in available_prompts:
+                raise ValueError(
+                    f"Prompt with name '{key}' does not exist in the simulator {self.name}. Use get_prompts() to see available prompts."
+                )
+            if not isinstance(value, Prompt):
+                raise ValueError(
+                    f"Prompt with name '{key}' must be an instance of 'Prompt'"
```

Comment:
```suggestion
                    f"Prompt with name '{key}' must be an instance of 'ragas.prompt.PydanticPrompt'"
```

Reply:
I see, thanks I didn't know that this is the use of Mixins.